homeassistant/components/advantage_air/climate.py

```diff
@@ -130,7 +125,22 @@ def hvac_mode(self):
     @property
     def hvac_modes(self):
         """Return the supported HVAC modes."""
-        return AC_HVAC_MODES
+        if self._ac.get("myAutoModeEnabled"):
+            return [
+                HVAC_MODE_OFF,
+                HVAC_MODE_COOL,
+                HVAC_MODE_HEAT,
+                HVAC_MODE_FAN_ONLY,
+                HVAC_MODE_DRY,
+                HVAC_MODE_AUTO,
+            ]
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_FAN_ONLY,
+            HVAC_MODE_DRY,
+        ]
```

Comment:
I would prefer this as a single base list that HVAC_MODE_AUTO is appended to if it's available, instead of having two copies that need to be kept in sync.

Reply:
@viiru- I have pushed a change to address this.

`if self._ac.get("myAutoModeEnabled"):
    return AC_HVAC_MODES + [HVAC_MODE_AUTO]
return AC_HVAC_MODES`